### DISCUSSION TITLE
Expose Buttons in the OAuth Sign In Control on the Developer Portal

### DISCUSSION THREAD

**Discussion by nimccoll** [Azure/api-management-developer-portal#2592] [Feature request]
Expose each button on the OAuth Sign In control on the Developer Portal as a separate control that can be styled individually including hiding a button if necessary.

The OAuth Sign In control appears as a single control in the Developer Portal designer experience. You cannot see or style the individual buttons that will be included in the control. Expose these buttons so they can be individually styled or hidden as needed without having to resort to self-hosting the Developer Portal.

Customer has no control over the look and feel of the buttons that are rendered for each of the sign in options on the OAuth Sign In control. This prevents them for highlighting one button as the primary or hiding a button that they do not want used from the Developer Portal, but are using from APIs.

Customer has explored the use of delegated sign in, but that requires the creation and maintenance of a separate site to host the sign in experience.

**Comment by ghost** [comment]
@nimccoll, thank you for opening this issue. We will triage it within the next few business days.

**Comment by ghost** [comment]
@nimccoll, thank you for requesting this feature.

**Comment by mikebudzynski** [comment]
@malincrist - FYI.
